crates/wascc-provider/src/states/running.rs

```diff
@@ -11,8 +11,13 @@ use kubelet::state::prelude::*;
 pub struct Running;
 
 #[async_trait::async_trait]
-impl State<PodState> for Running {
-    async fn next(self: Box<Self>, _pod_state: &mut PodState, _pod: &Pod) -> Transition<PodState> {
+impl State<ProviderState, PodState> for Running {
```

Comment:
Still thinking through how we could do this, but having two states like this in a type feels overly confusing. Is there a way we can condense these to a single `SharedState` type or the like? My thinking is that the state machine shouldn't care about which shared state is coming from where. It should just have one shared state that has everything the state machine itself needs